grafana/resource_machine_learning_outlier.go

```diff
@@ -0,0 +1,268 @@
+package grafana
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/grafana/machine-learning-go-client/mlapi"
+)
+
+func ResourceMachineLearningOutlierDetector() *schema.Resource {
+	return &schema.Resource{
+
+		Description: `
+An outlier detector monitors the results of a query and reports when its values are outside normal bands.
+
+The normal band is configured by choice of algorithm, its sensitivity and other configuration.
+
+Visit https://grafana.com/docs/grafana-cloud/machine-learning/outlier-detection/ for mroe details.
+`,
+
+		CreateContext: resourceMachineLearningOutlierCreate,
+		ReadContext:   resourceMachineLearningOutlierRead,
+		UpdateContext: resourceMachineLearningOutlierUpdate,
+		DeleteContext: resourceMachineLearningOutlierDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Description: "The ID of the outlier detector.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"name": {
+				Description: "The name of the outlier detector.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"metric": {
+				Description: "The metric used to query the outlier detector results.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"description": {
+				Description: "A description of the outlier detector.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"datasource_id": {
+				Description: "The id of the datasource to query.",
+				Type:        schema.TypeInt,
+				Optional:    true,
+			},
+			"datasource_uid": {
+				Description: "The uid of the datasource to query.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"datasource_type": {
+				Description: "The type of datasource being queried. Currently allowed values are prometheus, graphite, loki, postgres, and datadog.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
```

Comment:
Can you add a `ValidateFunc` for this one? You can use [`validation.StringInSlice`](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation#StringInSlice) for it.

Reply:
Good idea, thanks. Did same for algorithm name too.